driver/driver.go

```diff
@@ -40,9 +65,28 @@ type Module interface {
 	Close() error
 }
 
+type ParseOptions struct {
+	Mode     Mode
+	Language string
+	Filename string
+}
+
 // Driver is an interface for a language driver that returns UAST.
 type Driver interface {
-	Parse(ctx context.Context, mode Mode, src string) (nodes.Node, error)
+	// Parse reads the input string and constructs an AST representation of it.
+	//
+	// Language can be specified by providing ParseOptions. If the language is not set,
+	// it will be set during the Parse call if implementation supports language detection.
+	//
+	// Depending on the mode, AST may be transformed to different UAST variants.
+	// ErrModeNotSupported is returned for unsupported transformation modes.
+	//
+	// Syntax errors are indicated by returning ErrSyntax.
+	// In this case a non-empty UAST may be returned, if driver supports partial parsing.
+	//
+	// Native driver failures are indicated by ErrDriverFailure and UAST transformation are indicated by ErrTransformFailure.
+	// All other errors indicate a protocol or server failure.
+	Parse(ctx context.Context, src string, opts *ParseOptions) (nodes.Node, error)
```

Comment:
Could someone please help me understand the impact of this change: what else needs to be changed and where?

Reply:
This only affects an SDK itself. Drivers will also need to be updated, but only to bump version. They just call to SDK that creates and hosts those `Driver`s, thus no other changes are necessary. So I think it's safe to assume it's a minor change.

Reply:
Regarding "taking advantage" part, the `client-go.v3` needs to be updated (already have the change, but it's blocked by this). Again, only internals of the client will change. User-facing API will not be affected.